src/aks-preview/azext_aks_preview/agentpool_decorator.py

```diff
@@ -1070,6 +1100,14 @@ def set_up_virtual_machines_profile(self, agentpool: AgentPool) -> AgentPool:
 
         return agentpool
 
+    def set_up_localdns_profile(self, agentpool: AgentPool) -> AgentPool:
+        """Set up local DNS profile for the AgentPool object if provided via --localdns-config."""
+        self._ensure_agentpool(agentpool)
+        localdns_profile = self.context.get_localdns_profile()
+        if localdns_profile is not None:
+            agentpool.local_dns_profile = self.models.LocalDNSProfile(**localdns_profile)
```

Comment:
If that's your intention, I recommend performing the conversion (from dict to model) within get_localdns_profile. Just be mindful of error handling, as user input may be invalid.

However, if you prefer not to validate the keys (i.e., comparing what's supported by the model versus what's provided by user input) on the client side, you can skip the conversion.